To: vim_dev@googlegroups.com
Subject: Patch 7.3.087
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.087
Problem:    EINTR is not always defined.
Solution:   Include errno.h in vim.h.
Files:	    src/if_cscope.c, src/if_tcl.c, src/integration.c, src/memline.c,
	    src/os_mswin.c, src/os_win16.c, src/os_win32.c, src/vim.h,
	    src/workshop.c


*** ../vim-7.3.086/src/if_cscope.c	2010-12-17 18:06:00.000000000 +0100
--- src/if_cscope.c	2010-12-17 20:06:01.000000000 +0100
***************
*** 13,20 ****
  
  #if defined(FEAT_CSCOPE) || defined(PROTO)
  
- #include <string.h>
- #include <errno.h>
  #include <assert.h>
  #include <sys/types.h>
  #include <sys/stat.h>
--- 13,18 ----
*** ../vim-7.3.086/src/if_tcl.c	2010-08-15 21:57:27.000000000 +0200
--- src/if_tcl.c	2010-12-17 20:06:56.000000000 +0100
***************
*** 74,80 ****
  #endif
  
  #include <tcl.h>
- #include <errno.h>
  #include <string.h>
  
  typedef struct
--- 74,79 ----
*** ../vim-7.3.086/src/integration.c	2010-08-15 21:57:29.000000000 +0200
--- src/integration.c	2010-12-17 20:07:12.000000000 +0100
***************
*** 33,39 ****
  #include <sys/un.h>
  #endif
  
- #include <errno.h>
  #include <sys/types.h>
  #include <sys/socket.h>
  #include <sys/param.h>
--- 33,38 ----
*** ../vim-7.3.086/src/memline.c	2010-12-17 18:06:00.000000000 +0100
--- src/memline.c	2010-12-17 20:07:31.000000000 +0100
***************
*** 52,61 ****
  # include <proto/dos.h>	    /* for Open() and Close() */
  #endif
  
- #ifdef HAVE_ERRNO_H
- # include <errno.h>
- #endif
- 
  typedef struct block0		ZERO_BL;    /* contents of the first block */
  typedef struct pointer_block	PTR_BL;	    /* contents of a pointer block */
  typedef struct data_block	DATA_BL;    /* contents of a data block */
--- 52,57 ----
*** ../vim-7.3.086/src/os_mswin.c	2010-12-17 18:06:00.000000000 +0100
--- src/os_mswin.c	2010-12-17 20:08:14.000000000 +0100
***************
*** 30,36 ****
  # include <string.h>
  #endif
  #include <sys/types.h>
- #include <errno.h>
  #include <signal.h>
  #include <limits.h>
  #include <process.h>
--- 30,35 ----
*** ../vim-7.3.086/src/os_win16.c	2010-12-17 18:06:00.000000000 +0100
--- src/os_win16.c	2010-12-17 20:08:53.000000000 +0100
***************
*** 25,31 ****
  #include <dos.h>
  #include <string.h>
  #include <sys/types.h>
- #include <errno.h>
  #include <signal.h>
  #include <limits.h>
  #include <process.h>
--- 25,30 ----
*** ../vim-7.3.086/src/os_win32.c	2010-12-17 18:06:00.000000000 +0100
--- src/os_win32.c	2010-12-17 20:09:18.000000000 +0100
***************
*** 27,33 ****
  #endif
  
  #include <sys/types.h>
- #include <errno.h>
  #include <signal.h>
  #include <limits.h>
  #include <process.h>
--- 27,32 ----
*** ../vim-7.3.086/src/vim.h	2010-12-17 18:06:00.000000000 +0100
--- src/vim.h	2010-12-17 20:09:56.000000000 +0100
***************
*** 480,485 ****
--- 480,490 ----
  # include <sys/stat.h>
  #endif
  
+ #if defined(HAVE_ERRNO_H) || defined(DJGPP) || defined(WIN16) \
+ 	|| defined(WIN32) || defined(_WIN64) || defined(__EMX__)
+ # include <errno.h>
+ #endif
+ 
  /*
   * Allow other (non-unix) systems to configure themselves now
   * These are also in os_unix.h, because osdef.sh needs them there.
*** ../vim-7.3.086/src/workshop.c	2010-08-15 21:57:26.000000000 +0200
--- src/workshop.c	2010-12-17 20:09:31.000000000 +0100
***************
*** 16,22 ****
  #include <sys/types.h>
  #include <netdb.h>
  #include <netinet/in.h>
- #include <errno.h>
  #include <sys/socket.h>
  #ifdef HAVE_LIBGEN_H
  # include <libgen.h>
--- 16,21 ----
*** ../vim-7.3.086/src/version.c	2010-12-17 18:52:56.000000000 +0100
--- src/version.c	2010-12-17 20:14:49.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     87,
  /**/

-- 
How To Keep A Healthy Level Of Insanity:
17. When the money comes out the ATM, scream "I won!, I won! 3rd
    time this week!!!!!"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
